--- a/aws_kms_pkcs11.cpp	2026-02-06 10:22:49.940896965 -0500
+++ b/aws_kms_pkcs11.cpp	2026-02-06 13:55:04.501821983 -0500
@@ -405,11 +405,13 @@
     if (session == NULL) {
         return CKR_SESSION_HANDLE_INVALID;
     }
-    for (auto it = active_sessions->begin(); it != active_sessions->end(); ) {
-        if (*it == session) {
-            active_sessions->erase(it);
-        } else {
-            it++;
+    if (active_sessions != NULL) {
+        for (auto it = active_sessions->begin(); it != active_sessions->end(); ) {
+            if (*it == session) {
+                active_sessions->erase(it);
+            } else {
+                it++;
+            }
         }
     }
     free(session);
